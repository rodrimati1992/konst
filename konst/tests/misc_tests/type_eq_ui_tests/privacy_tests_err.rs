#![allow(unused_imports)]

mod opt_pub_self {
    konst::polymorphism::type_eq_projection_fn!{
        pub(self) fn project_opt => Option<from T>
    }
}

const _: () = {
    opt_pub_self::project_opt::<u8, u8>;
};


mod opt_pub_super {
    pub mod foo {
        konst::polymorphism::type_eq_projection_fn!{
            pub(super) fn project_opt => Option<from T>
        }
    }

    const _: () = {
        foo::project_opt::<u8, u8>;
    };
}

const _: () = {
    opt_pub_super::foo::project_opt::<u8, u8>;
};

fn main(){}