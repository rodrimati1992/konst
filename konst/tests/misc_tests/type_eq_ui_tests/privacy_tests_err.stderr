error[E0603]: function `project_opt` is private
  --> tests/misc_tests/type_eq_ui_tests/privacy_tests_err.rs:10:19
   |
10 |     opt_pub_self::project_opt::<u8, u8>;
   |                   ^^^^^^^^^^^ private function
   |
note: the function `project_opt` is defined here
  --> tests/misc_tests/type_eq_ui_tests/privacy_tests_err.rs:4:5
   |
4  | /     konst::polymorphism::type_eq_projection_fn!{
5  | |         pub(self) fn project_opt => Option<from T>
6  | |     }
   | |_____^
   = note: this error originates in the macro `konst::polymorphism::type_eq_projection_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: function `project_opt` is private
  --> tests/misc_tests/type_eq_ui_tests/privacy_tests_err.rs:27:25
   |
27 |     opt_pub_super::foo::project_opt::<u8, u8>;
   |                         ^^^^^^^^^^^ private function
   |
note: the function `project_opt` is defined here
  --> tests/misc_tests/type_eq_ui_tests/privacy_tests_err.rs:16:9
   |
16 | /         konst::polymorphism::type_eq_projection_fn!{
17 | |             pub(super) fn project_opt => Option<from T>
18 | |         }
   | |_________^
   = note: this error originates in the macro `konst::polymorphism::type_eq_projection_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
